crates/ruff/src/rules/pydocstyle/rules/blank_before_after_class.rs

```diff
@@ -228,8 +228,33 @@ pub(crate) fn blank_before_after_class(checker: &mut Checker, docstring: &Docstr
         }
 
         let mut lines = UniversalNewlineIterator::with_offset(after, after_range.start());
+        let first_line = lines.next();
+        let first_line_start = first_line.as_ref().map(|l| l.start()).unwrap_or_default();
+
+        // Edge case: The next statement after the docstring is on the same line separated by a
+        // semicolon.
+        if let Some(first_line) = &first_line {
+            if !first_line.trim().is_empty() {
+                let next_statement = after
+                    .trim_start()
+                    .strip_prefix(';')
+                    .expect("Another statement on the same line must be separated by a semicolon");
+                let indentation = indentation_at_offset(docstring.start(), checker.locator())
+                    .expect("Own line docstring must have indentation");
+                let mut diagnostic = Diagnostic::new(OneBlankLineAfterClass, docstring.range());
+                if checker.patch(diagnostic.kind.rule()) {
+                    let line_ending = checker.stylist().line_ending().as_str();
+                    diagnostic.set_fix(Fix::automatic(Edit::replacement(
+                        line_ending.to_string() + line_ending + indentation + next_statement,
+                        first_line_start,
+                        first_line.end(),
+                    )));
+                }
+                checker.diagnostics.push(diagnostic);
+                return;
+            }
+        }
```

Comment:
Could we move this up to line 223 where we intentionally skip the first line? 

We could then improve the perforamnce by cloning the univeral newlines iterator rather than creating a new one from scratch on line 230. That avoids that we scan the first line 2 (actually 3 times, 2 to find the end of the first line, once to trim it.)

Reply:
Turns out we do actually have to parse the first line twice